nncf/quantization/layers.py

```diff
@@ -230,17 +240,40 @@ def __init__(self, config):
         self.collect_scale_statistics = False
         if self.per_channel:
             self.scale_shape = get_per_channel_scale_shape(self.input_shape, self.is_weights)
-        self.scale = nn.Parameter(torch.ones(self.scale_shape), requires_grad=True)
-        self.eps = 1e-16
+        self.scale_tensor = nn.Parameter(torch.ones(self.scale_shape), requires_grad=True)
+        if self.scale_log_flag:
+            self.scale_tensor.data.log_()
+            self.eps = 0
+        else:
+            self.eps = 1e-16
         if config.signedness_to_force is not None:
             self.signed = int(config.signedness_to_force)
         self.set_level_ranges()
 
+        #register hooks to convert scale from log vertion to original vertion during load and save
+        def hook_fn_load_state_dict(state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs):
+            v = state_dict.pop(prefix + 'scale')
+            if self.scale_log_flag:
+                v = v.abs().log().detach()
+            state_dict[prefix + 'scale_tensor'] =  v
+        self.load_state_dict_hook = HookAutoRemove(self._register_load_state_dict_pre_hook(hook_fn_load_state_dict))
+
+        def hook_fn_state_dict(module, destination, prefix, local_metadata):
+            v = destination.pop(prefix + 'scale_tensor')
+            if module.scale_log_flag:
+                v = v.exp().detach()
+            destination[prefix + 'scale'] = v
+        self.state_dict_hook = HookAutoRemove(self._register_state_dict_hook(hook_fn_state_dict))
+
+    @property
+    def scale(self):
+        return self.scale_tensor.exp() if self.scale_log_flag else self.scale_tensor
```

Comment:
Should define a setter as well.

Reply:
I can define setter but it is not clear what input should be for this setter. And where it will be used.

Reply:
ok. I prepare and commit some variant for scale. if it is ok lets keep it. But still have no idea why it is needed.